.github/workflows/ci.yml

```diff
@@ -328,6 +320,23 @@ jobs:
           # fixed to avoid triggering false positive; see https://github.com/golangci/golangci-lint-action/issues/535
           version: v1.52.2
           working-directory: clients/go
+  go-apidiff:
+    name: Go Backward Compatibility
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 #we need to fetch stable/* branches in order to be run compatibility checks
```

Comment:
This checkout takes ~19s while a direct ref checkout takes ~3s.

🔧 We could consider improving this by checking out using defaults (only the direct ref), and additionally fetching `origin/stable/8.2` in a step before running the `go-apidiff` step 🌱

Reply:
Great idea. Will look into it